astropy/utils/data.py

```diff
@@ -2046,6 +2046,13 @@ def _rmtree(path, replace=None):
                 )
             )
             raise
+        except OSError as e:
+            if e.errno == errno.EXDEV:
+                warn(e.strerror, AstropyWarning)
+                shutil.move(path, os.path.join(d, "to-zap"))
+            else:
```

Comment:
Now I see that the original code did not raise on OSError but perhaps it was an oversight. I do not see why it should be ignored.